docs/guides/moving-data.md

```diff
@@ -0,0 +1,325 @@
+---
+description: Moving data to and from cloud providers
+tags:
+    - data
+    - cloud providers
+    - AWS
+    - S3
+    - Azure Storage
+    - Azure Blob Storage
+    - Azure
+    - GCP
+    - Google Cloud Storage
+    - GCS
+    - moving data
+search:
+  boost: 2
+---
+
+# How to move data to and from cloud providers
+
+Sending data to cloud-based storage and retrieving data from that storage is a common task in data engineering.
+In this guide we'll learn how to use Prefect to move data to and from AWS, Azure, and GCP blob storage.
+
+## Prerequisites
+
+- Prefect [installed](/getting-started/installation/)
+- Authenticated with [Prefect Cloud](/cloud/cloud-quickstart/) (or self-hosted [Prefect server](/guides/host/) instance)
+- A cloud provider account (e.g. [AWS](https://aws.amazon.com/))
+
+## Install relevant Prefect integration library
+
+In the CLI, install the Prefect integration library for your cloud provider:
+
+=== "AWS"
+
+    [prefect-aws](https://prefecthq.github.io/prefect-aws/) provides blocks for interacting with AWS services.
+
+    ```bash
+    pip install -U prefect-aws
+    ```
+
+=== "Azure"
+
+    [prefect-azure](https://prefecthq.github.io/prefect-azure/) provides blocks for interacting with Azure services.
+
+    ```bash
+     pip install -U prefect-azure
+    ```
+
+=== "GCP"
+
+    [prefect-gcp](https://prefecthq.github.io/prefect-gcp/) provides blocks for interacting with GCP services.
+    
+    ```bash
+     pip install -U prefect-gcp
+    ```
+
+## Register the block types
+
+Register the new block types with Prefect Cloud (or with your self-hosted Prefect server instance):
+
+=== "AWS"
+
+    ```bash
+    prefect block register -m prefect_aws  
+    ```
+
+=== "Azure"
+
+    ```bash
+    prefect block register -m prefect_azure 
+    ```
+
+=== "GCP"
+
+    ```bash
+    prefect block register -m prefect_gcp
+    ```
+
+We should see a message in the CLI that several block types were registered.
+If we check the UI, we should see the new block types listed.
+
+## Create a storage bucket
+
+Create a storage bucket in the cloud provider account.
+Ensure the bucket is publicly accessible or create a user or service account with the appropriate permissions to fetch and write data to the bucket.
+
+## Create a credentials block
+
+If the bucket is private, there are several options to authenticate:
+
+1. Use a worker that is authenticated at deployment run time.
```

Comment:
this is a little more subtle - the worker doesn't need to be authenticated, it's the deployment's runtime that needs the authentication (and just authenticating the worker doesn't guarantee the deployment's runtime will inherit that)

Reply:
Thank you. Updated.